teamcity/resource_build_config.go

```diff
@@ -630,6 +675,62 @@ func flattenBuildStepCmdLine(s *api.StepCommandLine) map[string]interface{} {
 	return m
 }
 
+func flattenBuildFeatures(bfs []api.BuildFeature) ([]map[string]interface{}, error) {
+	var bfsToSave []map[string]interface{}
+	for _, bf := range bfs {
+		bfToSave := make(map[string]interface{})
+		cbf := bf.(*api.CommonBuildFeature)
+		bfToSave["type"] = cbf.Type()
+
+		if cbf.Properties() != nil && cbf.Properties().Count > 0 {
+			bfToSave["properties"] = make([]map[string]string, 0)
```

Comment:
It's not necessary to `make` this entry, since `nil` is a valid (empty) slice.

You can do this instead:

```go
props := cbf.Properties()
if props != nil && props.Count > 0 {
    for _, prop := range props.Items {
        propMap := map[string]string{
            "name":  property.Name,
            "value": property.Value,
        }
        bfToSave["properties"] = append(bfToSave["properties"], propMap)
    }
}
```